### ISSUE TITLE
The content type application/json+hal causes a NoTransformationFoundException

### ISSUE THREAD

**Issue by rekire** [ktorio/ktor#1472]
**Ktor client 1.2.6 and with OkHttp Engine and GsonSerializer**
Used dependencies:
```
compile "io.ktor:ktor-client-okhttp:1.2.6"
compile "io.ktor:ktor-client-json:1.2.6"
compile "io.ktor:ktor-client-gson:1.2.6"
```

**Describe the bug**
The content type application/json+hal causes a NoTransformationFoundException

**To Reproduce**
Create a REST request which returns an data class when the response has the content type `application/json+hal`.  
More details and a bit too [complex example](https://stackoverflow.com/q/59221697/995926) at Stack Overflow

**Expected behavior**
I would expect that the ktor behaves the same with the content type `application/json` and just returns my data class.

**Screenshots**
```
Exception in thread "main" io.ktor.client.call.NoTransformationFoundException: No transformation found: class kotlinx.coroutines.io.ByteBufferChannel -> class my.package.name.Skills
        at io.ktor.client.call.HttpClientCall.receive(HttpClientCall.kt:88)
        at my.package.name.AmazonApi$fetchSkills$1.invokeSuspend(AmazonApi.kt:200)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(PipelineContext.kt:215)
        at io.ktor.util.pipeline.SuspendFunctionGun.loop(PipelineContext.kt:172)
        at io.ktor.util.pipeline.SuspendFunctionGun.access$loop(PipelineContext.kt:67)
        at io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(PipelineContext.kt:122)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
        at io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(PipelineContext.kt:215)
        at io.ktor.util.pipeline.SuspendFunctionGun.loop(PipelineContext.kt:172)
        at io.ktor.util.pipeline.SuspendFunctionGun.access$loop(PipelineContext.kt:67)
        at io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(PipelineContext.kt:122)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
        at io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(PipelineContext.kt:215)
        at io.ktor.util.pipeline.SuspendFunctionGun.loop(PipelineContext.kt:172)
        at io.ktor.util.pipeline.SuspendFunctionGun.access$loop(PipelineContext.kt:67)
        at io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(PipelineContext.kt:122)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
        at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:241)
        at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:270)
        at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:79)
        at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:54)
        at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
        at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:36)
        at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
        at my.package.name.AmazonApi.fetchSkills(AmazonApi.kt:131)
        at my.package.name.Cli.createProject(Cli.kt:305)
        at my.package.name.Cli.parseArgs(Cli.kt:208)
        at my.package.name.Cli$Companion.main(Cli.kt:508)
        at my.package.name.Cli.main(Cli.kt)
```

**Comment by cy6erGn0m** [comment]
`json+hal` content type should be added to the accept list for JsonFeature

**Comment by rekire** [comment]
Can you give me a code snipit how this should look like?

I'm using this:

    val httpClient = HttpClient(OkHttp) {
        install(JsonFeature) {
            serializer = GsonSerializer()
        }
    }

**Comment by rekire** [comment]
Based on @cy6erGn0m's comment I was able to solve the problem by using this:

    val httpClient = HttpClient(OkHttp) {
        install(JsonFeature) {
            serializer = GsonSerializer()
            acceptContentTypes += ContentType("application","json+hal")
        }
    }

However this is causing two warnings, how can I make it better?

**Comment by e5l** [comment]
It looks like it is a single right way to do it.

**Comment by cy6erGn0m** [comment]
@e5l perhaps we need a builder function `accept(ContentType)` instead and hide the property. + add content type constant `Application.JsonHal`. What do you think?

**Comment by e5l** [comment]
I'm sure we can.

**Comment by dmitrievanthony** [comment]
Hi @rekire, I merged the fix that will allow you to configure `HalJson` content type a bit easier. With this fix your code will look like the following example:

```
install(JsonFeature) {
    serializer = GsonSerializer()
    accept(ContentType.Application.HalJson)
}
```

By the way, during the PR discussion, we have figured out that that the content type should be called `application/hal+json`, not `application/json+hal` according to the standard.  Be careful with it. If the API you use responses with `application/json+hal` you'll have to specify it manually:

```
install(JsonFeature) {
    serializer = GsonSerializer()
    accept(ContentType("application","json+hal"))
}
```

I hope it will help you!
